tests/integration/test_analysisd/conftest.py

```diff
@@ -38,3 +44,37 @@ def callback_analysisd_startup(line):
 
     log_monitor = FileMonitor(LOG_FILE_PATH)
     log_monitor.start(timeout=30, callback=callback_analysisd_startup)
+
+
+@pytest.fixture(scope='module')
+def configure_custom_rules(request, get_configuration):
+    """Configure a syscollector custom rules for testing.
+    Restarting wazuh-analysisd is required to apply this changes.
+    """
+    data_dir = getattr(request.module, 'data_dir')
+    source_rule = os.path.join(data_dir, get_configuration['rule_file'])
+    target_rule = os.path.join(CUSTOM_RULES_PATH, get_configuration['rule_file'])
+
+    # copy custom rule with specific privileges
+    shutil.copy(source_rule, target_rule)
+    shutil.chown(target_rule, WAZUH_UNIX_USER, WAZUH_UNIX_GROUP)
+
+    yield
+
+    # remove custom rule
+    os.remove(target_rule)
+
+
+@pytest.fixture(scope='module')
+def restart_analysisd():
+    """Restart analysisd and truncate logs."""
+
+    truncate_file(ALERT_FILE_PATH)
+    truncate_file(LOG_FILE_PATH)
+
+    control_service('restart', daemon=ANALYSISD_DAEMON)
+    check_daemon_status(running_condition=True, target_daemon=ANALYSISD_DAEMON)
```

Comment:
This may not be interesting, because sometimes, you want to restart analysisd to know that the configuration is wrong.